distribution/lib/Standard/Visualization/0.0.0-dev/src/Table.enso

```diff
@@ -0,0 +1,222 @@
+from Standard.Base import all
+
+from Standard.Base.Data.Text.Extensions import slice_text
+from Standard.Visualization.Text import get_chunk_from_line
+import Standard.Base.Error.Common.Not_Found
+from Standard.Table import Table
+
+
+## PRIVATE
+
+   Update about the table layout to be sent to the GUI. Updates can be incremental and only contain
+   information about some rows or columns. Even updating information is fine and will be handled
+   by the UI.
+type TableSpecificationUpdate
+
+    Value row_heights column_widths column_names row_names
+
+
+## PRIVATE
+
+   Message to be sent to the IDE. Contains information about the table content and the table layout.
+   The table content is encoded as chunks of text, together with their respective coordinate in the
+   table. Coordinates have a first component that indicates row/column and a second component that
+   indicates the chunk index and line.
+type TableUpdate
+
+    Value chunks table_specification_update
+
+    ## PRIVATE
+
+       Generate JSON that can be consumed by the visualization.
+    to_json : Text
+    to_json self =
+        chunks = ["chunks", self.chunks]
+        table_specification_update = ["table_specification_update", self.table_specification_update]
+        Json.from_pairs [chunks, table_specification_update] + ""
```

Comment:
Currently, it does mind. Either we have some extra boilerplate here, to remove the extra content, or we need to add it to the GUI to ignore it. Since we send this over the network, this seems more sensible.